org.civicrm.reqangular/test/decorators/ui-bootstrap/uib-modal.decorator.spec.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/angularMocks',
+  'common/angular',
+  'common/decorators/ui-bootstrap/uib-modal.decorator',
+  'common/angularBootstrap'
+], function (_, angularMocks, angular, $uibModalDecorator, $document) {
+  'use strict';
+
+  describe('$uibModal.open', function () {
+    var $uibModal, $provide, $document;
+
+    beforeEach(module('ui.bootstrap'));
+
+    beforeEach(function () {
+      module(function (_$provide_) {
+        $provide = _$provide_;
+      });
+      inject(function () {
+        $provide.decorator('$uibModal', $uibModalDecorator);
+      });
+    });
+
+    beforeEach(inject(function (_$uibModal_, _$document_) {
+      $uibModal = _$uibModal_;
+      $document = _$document_;
+    }));
+
+    describe('init', function () {
+      var arg1 = { template: '<component></component>' };
+      var arg2 = 'another_argument';
+      var elements = {};
+      var doc, originalStyles;
+
+      beforeEach(function () {
+        doc = $document[0];
+        elements.body = doc.body;
+        elements.html = doc.getElementsByTagName('html')[0];
+
+        if (!originalStyles) {
+          originalStyles = {
```

Comment:
shouldn't you mirror what you do in the decorator, having the original style directly as a property of the element? This looks inconsistent

Reply:
I don't think the implementation must be exactly the same as tests. Test is doing exactly what it is designed for and the implementation produces the same results but differently. If we are testing the implementation in the exactly same way we have a risk of writing wrong tests (if we wrote wrong implementation).